src/fairseq2/nn/utils/fairseq1_mask.py

```diff
@@ -0,0 +1,147 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Optional, Tuple
+
+import numpy as np
+import torch
+
+
+def compute_mask_indices(
```

Comment:
we need to review the below code and make it more readeable and efficient

Reply:
unify pure python and numpy/torch usage

Reply:
let's add some unittests first on it, and next optimize if time allows

Reply:
Made obsolete through Can's refactoring.